src/python/pants/jvm/resolve/coursier_fetch_integration_test.py

```diff
@@ -576,6 +576,44 @@ def test_fetch_one_coord_with_mismatched_coord(rule_runner: RuleRunner) -> None:
         rule_runner.request(ClasspathEntry, [lockfile_entry])
 
 
+@maybe_skip_jdk_test
+def test_fetch_one_coord_with_non_jar_packaging(rule_runner: RuleRunner) -> None:
+    """This test demonstrates that fetch_one_coord is able to download non-jar artifacts such as
+    protoc plugin binaries that are distribured via Maven Central."""
+    coordinate = Coordinate(
+        group="io.grpc",
+        artifact="protoc-gen-grpc-java",
+        version="1.48.0",
+        packaging="exe",
+        classifier="linux-x86_64",
+    )
+    resolved_lockfile = rule_runner.request(
+        CoursierResolvedLockfile,
+        [ArtifactRequirements.from_coordinates([coordinate])],
+    )
+
+    entries = [
+        e
+        for e in resolved_lockfile.entries
+        if e.coord
+        == Coordinate(
+            group="io.grpc",
+            artifact="protoc-gen-grpc-java",
+            version="1.48.0",
+            packaging="exe",
+            classifier="linux-x86_64",
+            strict=True,
+        )
+    ]
+    assert len(entries) == 1
+    entry = entries[0]
+
+    classpath_entry = rule_runner.request(ClasspathEntry, [entry])
```

Comment:
The effect of this change is that we will have `ClasspathEntry`s which are not `JARs`. Would be good to update the `ClasspathEntry` docs to account for that possibility.

Are they supposed to be laid out a particular way on the classpath (inside of a loose directory?) for loading?

Reply:
> Are they supposed to be laid out a particular way on the classpath (inside of a loose directory?) for loading?

Not that I know of. The "exe" packaging means it will just be a free-standing executable binary on the classpath somewhere (assuming there are jar artifacts that have the "exe" as a transitive dependency).

At least for java-grpc support in https://github.com/pantsbuild/pants/pull/16350, that is not how I envision this support being used in practice. In that case, the protobuf codegen backend makes a custom `Digest` using the "exe"'s `FileDigest` and does not keep the file in the form of the classpath.

Reply:
Added a comment with a TODO about filtering.